packages/venia-ui/lib/components/CreateAccountPage/createAccountPage.js

```diff
@@ -1,40 +1,43 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { useCallback, useMemo } from 'react';
+import { func, shape } from 'prop-types';
 import { withRouter } from '@magento/venia-drivers';
 import { compose } from 'redux';
 import CreateAccountForm from '../CreateAccount';
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import defaultClasses from './createAccountPage.css';
 import { getCreateAccountInitialValues } from './helpers';
 
-class CreateAccountPage extends Component {
-    static propTypes = {
-        createAccount: PropTypes.func,
-        initialValues: PropTypes.shape({}),
-        history: PropTypes.shape({})
-    };
+const CreateAccountPage = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const { createAccount, history } = props;
 
-    createAccount = accountInfo => {
-        const { createAccount, history } = this.props;
-        createAccount({ accountInfo, history });
-    };
+    const handleCreateAccount = useCallback(
+        async accountInfo => {
+            await createAccount(accountInfo);
```

Comment:
Okay so this pattern is different - we wait for actions / asyncActions directly in the component.

Reply:
Yea, if there's a UI related update that should happen in the view not in the action.

Most awaited function calls in the actions are just dispatches or non-ui related things like updating storage.

Reply:
Ok, good to know, thanks 👍